### DISCUSSION TITLE
Welcome to tidal-connect Discussions!

### DISCUSSION THREAD

**Discussion by GioF71** [GioF71/tidal-connect#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Spoker1973** [comment]
Hi GioF71
First of all thank you for tidal connect new method.
I would like to note because you may not be aware that at the moment tidal does not allow you to listen to songs in FLAC format. Only MQA works and every other track shows quality as HIGH . The moment you start a track in FLAC format the quality immediately changes to HIGH and tidal does not play anything.
Do you perhaps know about this problem?

**Comment by GioF71** [comment]
Hello @Spoker1973, just two days ago I upgraded my subscription to "Hi Fi Plus", and noticed the same behavior.
When using the Tidal Connect container, I often see the sampling rate go to 96kHz, but I never saw 192kHz, even for albums labeled as "max" quality. So I concluded that either I have not been able to really find an album which is available in flac hi-res, or that the software included in the Tidal Connect image simply does not support this new streaming mode.
Also, when using the Tidal Plugin for Upmpdcli (see [here](https://github.com/GioF71/upmpdcli-docker/discussions/281)), I saw sampling rate up to a max of 48kHz: I concluded that mqa version was selected, but of course there is no mqa unfolding when using this solution.
Until the situation changes, I will probably revert back to the HiFi subscription. I don't have any MQA-enabled dac, and don't plan to buy one soon. Maybe, when I will hit the market for a new dac, MQA will have disappeared or will have become even less relevant than it is today.
I really hope Tidal will finally get rid of albums in MQA format, in favor of true hi-res releases.

**Comment by GioF71** [comment]
Thinking again, I did not experience a playback stop when trying to play "true" hires content. Can you indicate me the album you used, so I can try?
Also please post your configuration (the .env file).
Thank you.

**Reply by Spoker1973** [reply]
Thank you for your reply, if you mean songs in FLAC format then type into tidal search engine: Hi-Res FLAC - 24bit/192kHz and there you can find 69 flac hi res songs. If at that moment you look at the device on which you play tidal then you will see for a moment written  FLAC and immediately it will change to HIFI and the song will not play.  I am not an experienced Linux user, so please tell me how to call the configuration file from under SSH

**Comment by Spoker1973** [comment]
![Screenshot_20231006-131357_TIDAL](https://github.com/GioF71/tidal-connect/assets/147102072/0b43016c-9263-4e2f-9e05-8e0c3e6d6311)

**Comment by GioF71** [comment]
Everything plays on my system, but I confirm that nothing goes beyond 96kHz and I get the "MQA" writing instead of flac on the Tidal App. So the Tidal Connect app is doing the first unfolding, and it is playing the MQA version of the track.
One thing I remember when I upgraded the subscription, I had to logoff/logon on the android app, can you try doing this also?
And did you set "Wi-Fi streaming" to "Max" on the Audio & Quality / Quality preferences?
Also I generally disable "Normalize volume", but I don't believe if it is strictly relevant for this issue.

About your configuration, from your terminal just type:

```
cat $HOME/tidal-connect/.env
```

Thank you

**Reply by Spoker1973** [reply]
I didn`t try logout and then log in because i gave HI-FI + always, also i set up wi-fi streaming to max.

**Comment by Spoker1973** [comment]
env conf:
pawel@moode:~ $ cat $HOME/tidal-connect/.env
FRIENDLY_NAME=Desktop
MODEL_NAME=Raspberry Pi
CARD_NAME=AUDIO

**Comment by GioF71** [comment]
The config is ok... I would really try to logoff/logon again.
Which DAC are you using?
Can you post the output of

`cat /proc/asound/cards`

and 

`aplay -l
`

**Reply by Spoker1973** [reply]
I use SMSL SU-9 as DAC, when i connect dac via my PC then everything is ok, so i am going to try to logoff and then login again.
pawel@moode:~ $ cat /proc/asound/cards
 0 [b1             ]: bcm2835_hdmi - bcm2835 HDMI 1
                      bcm2835 HDMI 1
 1 [Headphones     ]: bcm2835_headpho - bcm2835 Headphones
                      bcm2835 Headphones
 2 [AUDIO          ]: USB-Audio - SMSL USB AUDIO
                      SMSL SMSL USB AUDIO at usb-0000:01:00.0-1.1, high speed
and:

pawel@moode:~ $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: b1 [bcm2835 HDMI 1], device 0: bcm2835 HDMI 1 [bcm2835 HDMI 1]
  Subdevices: 4/4
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
card 0: b1 [bcm2835 HDMI 1], device 1: bcm2835 HDMI 1 [bcm2835 HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Headphones [bcm2835 Headphones], device 0: bcm2835 Headphones [bcm2835 Headphones]
  Subdevices: 4/4
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
card 2: AUDIO [SMSL USB AUDIO], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**Comment by GioF71** [comment]
That traling space after "AUDIO" is suspicious.
Let's see which device is default with:

```
aplay -L
```

Do you have problem with everything or only with hi-res tracks?

Anyway maybe let's try with card index first:

```text
cd $HOME/tidal-connect
./configure.sh -i 2 -f "Desktop"
docker-compose up -d
```

**Reply by Spoker1973** [reply]
pawel@moode:~ $ aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
_audioout
_audioout__
alsaequal
plug_alsaequal
btstream
camilladsp
crossfeed
plug_bs2b
eqfa12p
plug_eqfa12p
invpolarity
trx_send
hw:CARD=b1,DEV=0
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Direct hardware device without any conversions
hw:CARD=b1,DEV=1
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Direct hardware device without any conversions
plughw:CARD=b1,DEV=0
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Hardware device with all software conversions
plughw:CARD=b1,DEV=1
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Hardware device with all software conversions
default:CARD=b1
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Default Audio Device
sysdefault:CARD=b1
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Default Audio Device
dmix:CARD=b1,DEV=0
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Direct sample mixing device
dmix:CARD=b1,DEV=1
    bcm2835 HDMI 1, bcm2835 HDMI 1
    Direct sample mixing device
hw:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Direct hardware device without any conversions
plughw:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Hardware device with all software conversions
default:CARD=Headphones
    bcm2835 Headphones, bcm2835 Headphones
    Default Audio Device
sysdefault:CARD=Headphones
    bcm2835 Headphones, bcm2835 Headphones
    Default Audio Device
dmix:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Direct sample mixing device
hw:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    Direct hardware device without any conversions
plughw:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    Hardware device with all software conversions
default:CARD=AUDIO
    SMSL USB AUDIO, USB Audio
    Default Audio Device
sysdefault:CARD=AUDIO
    SMSL USB AUDIO, USB Audio
    Default Audio Device
front:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    Front output / input
surround21:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    4.0 Surround output to Front and Rear speakers
surround41:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    IEC958 (S/PDIF) Digital Audio Output
dmix:CARD=AUDIO,DEV=0
    SMSL USB AUDIO, USB Audio
    Direct sample mixing device
pawel@moode:~ $

**Reply by Spoker1973** [reply]
I have no other problems just that one. I have tried to logout / login after that nothing had changed. When i try to play FLAC song i see short written FLAC and then it change to MQA and what i see on my DAC (look also at frequency):
![20231006_145513](https://github.com/GioF71/tidal-connect/assets/147102072/dde50e21-2509-4a0c-b6c2-7c7cf0557329)

**Comment by GioF71** [comment]
No there is probably no trailing space, sorry about that. Just post the logs of the container:

```
cd $HOME/tidal-connect
docker-compose logs
```

**Reply by Spoker1973** [reply]
pawel@moode:~/tidal-connect $ docker-compose logs
Attaching to tidal-connect
tidal-connect    | FRIENDLY_NAME=Desktop
tidal-connect    | MODEL_NAME=Audio Streamer
tidal-connect    | MQA_CODEC=false
tidal-connect    | MQA_PASSTHROUGH=false
tidal-connect    | CARD_NAME=NOT_SET
tidal-connect    | CARD_INDEX=2
tidal-connect    | Specified CARD_INDEX=[2]
tidal-connect    | Set card_index=[2]
tidal-connect    | Creating sound configuration file (card_index=2)...
tidal-connect    | Sound configuration file created.
tidal-connect    | defaults.pcm.card 2
tidal-connect    | pcm.!default {
tidal-connect    |   type plug
tidal-connect    |   slave.pcm hw
tidal-connect    | }
tidal-connect    | PLAYBACK_DEVICE=[default]
tidal-connect    | Starting Speaker Application in Background (TMUX)
tidal-connect    | Sleeping for a while (3 seconds)...
tidal-connect    | Starting TIDAL Connect ...
tidal-connect    | STARTING TidalConnect services, version: 1.1.3
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | ALSA lib pcm_dsnoop.c:618:(snd_pcm_dsnoop_open) unable to ope                                                                                                                                                             n slave
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.rear
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.center_lfe
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.side
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | Expression 'GetExactSampleRate( hwParams, &defaultSr )' faile                                                                                                                                                             d in 'src/hostapi/alsa/pa_linux_alsa.c', line: 895
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.hdmi
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.hdmi
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.modem
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.modem
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.phoneline
tidal-connect    | ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards                                                                                                                                                             .pcm.phoneline
tidal-connect    | Cannot connect to server socket err = No such file or directo                                                                                                                                                             ry
tidal-connect    | Cannot connect to server request channel
tidal-connect    | jack server is not running or cannot be started
tidal-connect    | JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for                                                                                                                                                              -1, skipping unlock
tidal-connect    | JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for                                                                                                                                                              -1, skipping unlock
tidal-connect    | STARTED TidalConnect services.

**Comment by Spoker1973** [comment]
unfortunately now i havent tidal connect, there is no device to choose at Tidal app

**Reply by GioF71** [reply]
Try restoring it with

```
cd $HOME/tidal-connect
docker-compose down
./configure.sh -n AUDIO -f Desktop
docker-compose up -d --force-recreate

```

**Reply by Spoker1973** [reply]
i restarted rasberry manually and Tidal started to work :)

**Comment by GioF71** [comment]
That looks consistent with what I see on my dac... your dac probably goes to 192kHz and indicates "MQA" because it can decode MQA. But it looks like that Tidal is giving you the MQA versions of the files instead of the hi-res flac files (non-mqa).
When I was mentioning that I was seeing "MQA", it was something displayed on the Android App, not on my DAC, which does not support MQA.
Also, considering your MQA-enabled DAC, you might play with the other two variables, `MQA_CODEC` and `MQA_PASSTHROUGH` which I cannot try by myself. That will probably not change the fact that the Tidal Connect app is receiving the MQA version of the files and not the full hires version.

Keep in mind that the underlying tidal-connect app is a closed-source software, it might not support at all the new streaming mode.

When you use your dac from your windows/mac desktop, does it behave differently? Sorry if I ask something you might have already expressed.

**Reply by Spoker1973** [reply]
I am aware that tidal-connect app is a closed source third party soft and might not support new streaming options.
And your second question when i use my DAC from Tidal on my desktop i behaves differently because i see FLAC instead of MQA :)
Now unfortunately something went wrong and tidal connect not works :(

**Comment by GioF71** [comment]
Did you try my previous post?

```
cd $HOME/tidal-connect
docker-compose down
./configure.sh -n AUDIO -f Desktop
docker-compose up -d --force-recreate

```

but maybe before that let's try and see if there is something in the logs:

`docker logs tidal-connect
`

**Comment by GioF71** [comment]
I see it works again, good to know. Unfortunately there seem to be limitations with this software, due to its origin.

**Reply by Spoker1973** [reply]
all is clear maybe in the future you`ll be able to solve that problem. Anyway - thank you so much for your collaboration.

**Reply by GioF71** [reply]
Honestly I doubt this can change. I instead believe that other projects have more potential:

https://github.com/tamland/python-tidal
https://github.com/tehkillerbee/mopidy-tidal

and I have built a [plugin for upmpdcli](https://github.com/GioF71/upmpdcli-docker/discussions/281) as I mentioned before. Still does not play flac hires though for some reasons.
But this Tidal Connect project is stuck where it is because, again, of its origin.

Anyway if you find it useful somehow, consider starring it :-)

**Comment by GioF71** [comment]
I just found this:

https://github.com/tamland/python-tidal/issues/188

"At the moment, Tidal only allows it's own apps to use the HiRes Flac files higher than 24bit 48kHz"

**Comment by GioF71** [comment]
And this:

https://github.com/orenskl/HiFiStreamer/issues/43

```
After some more digging it seems that currently TIDAL streams Hi Res FLAC only in 24bit/48Khz, further more this stream seems to be originated from the MQA version and it is not a true lossless version.

This will have to wait until we have a true Hi Res lossless stream
```

I guess we have to wait.

**Reply by Spoker1973** [reply]
Indeed we have to wait :)  If you have some news do not hesitate to let us know.

**Comment by Spoker1973** [comment]
Same problem have users of volumio, ifi zen stream and probably many more. So, iti is a global problem.

**Comment by looun** [comment]
With KODI addon TIDAL2 TIDAL MAX work 192@24bit.

**Comment by GioF71** [comment]
Yes, but that software, the python-tidal library and e.g. the upmpdcli tidal plugin which also relies on that library, are open source software while this Tidal Connect is not... unfortunately it cannot be updated by the community and there is nobody maintaining it. Or I'd better say, for sure there is somebody working on it, but they are maintaining and distributing updated versions for select oem vendors afaik

**Comment by looun** [comment]
should be need a new certificate, different from ifi zen.

**Reply by GioF71** [reply]
Ok... But how do we find such certificate?

**Comment by longday11** [comment]
@GioF71 
Thanks so much for this solution. 
I'm able to easily swap between my 2 USB dacs connected to api4 running moode.
I've played with LMS picoreplayer, bubble uonp and others. But it is so great just to use the native tidal interface. The tidal radio feature is great. 
Also, I set this up again 2 days ago and I swear there is a slight improvement in audio quality since last time (around a month ago). Might be placebo.
Thanks again. You are a legend!

**Comment by GioF71** [comment]
Thank you very much for your kind words!!
About the improved audio quality, I really don't know, but the application did not change. Might be placebo as you say.
Did you try [Mopidy-Tidal](https://github.com/GioF71/audio-tools/tree/main/players/mopidy-tidal), or the [Tidal plugin for upmpdcli](https://github.com/GioF71/audio-tools/tree/main/media-servers/tidal-hires)?

Please star the repo if you like the works, it is always encouraging :-)

**Comment by turambar517** [comment]
Hey thanks so much for the work on this! It's been great. I have it going using an XMOS sound device and a TAS5825M DAC.

**Comment by GioF71** [comment]
Thank you @turambar517!
Please star the repo if you like the work. This is always very encouraging.
Cheers!

**Comment by Spoker1973** [comment]
Hi guys, did you try to install tidal connect solution with moode audio version 9? Does it work?

**Reply by GioF71** [reply]
Hello, yes, it works for me and some other users with Moode Audio version 9.

**Reply by Spoker1973** [reply]
thank you for answer. As for me it works fantastic without problems. It`s a shame that doesn`t works with hi res files but maybe in the nearest furure :)

**Reply by GioF71** [reply]
Hello @Spoker1973, thank you for reporting back. Please star the repo if you like the work.
For now, we only have hires. We'll see what happens.
There are alternatives for hires, some fully opensource. Worth mentioning:

- Tidal Plugin for Upmpdcli lets you create a media server and then play to mpd+upmpdcli (offered by moode).
- Mopidy has a nice tidal plugin, you can install it using docker along with moode

Plus other options which cost only a few bucks, like BubbleUPnP and mConnect, with those again you can stream to mpd+upmpdcli

**Reply by Spoker1973** [reply]
Hello GioF71, thank you for comprehensive answer. Indeed i heard about alternatives but didn`t try to test it maybe except for one solution which is BubbleUPnP and mConnect. It works even fine but there is no native tidal connect :)
Do you have any tutorial how to install mopidy with moode? I am gonna to install new moode 9 and it would be good time to check how it works.

**Reply by GioF71** [reply]
> Hello GioF71, thank you for comprehensive answer. Indeed i heard about alternatives but didn`t try to test it maybe except for one solution which is BubbleUPnP and mConnect. It works even fine but there is no native tidal connect :) Do you have any tutorial how to install mopidy with moode? I am gonna to install new moode 9 and it would be good time to check how it works.

I don't have a tutorial, but I maintain a docker image for mopidy, which includes the [mopidy tidal plugin](https://github.com/tehkillerbee/mopidy-tidal).  
The docker image repository is [here](https://github.com/GioF71/mopidy-docker/tree/main).  
Look at the example docker-compose file in the README, it should work with a few modifications. Make sure you create the directories. I would just create a directory named "mopidy" inside the moode home directory, create the three directories `config`, `cache` and `data`.
Also check your group id and the audio group id using the following:

`id`

with this you get the uid of the current user. Ideally the one which you added to the docker group.
Then with

`getent group audio`

you can check if the `audio` group is `29` like in the sample docker-compose file. If it isn't, just correct it.
Run with a `docker-compose up -d` and you should be good to go. Monitor the logs with the usual `docker-compose logs -f`, then open the url of the moode box at the port 6680. Try to navigate to the tidal plugin, and the logs should present a URL you will need to follow. Authenticate with tidal and authorize the device. It should be all written in the README, but of course, feel free to report what might be missing and/or unclear.

**Reply by Spoker1973** [reply]
GioF71 thank you for massive help. Unfortunately i am not linux experinced user to edit files in order what you have writen, so i give up :)

**Reply by GioF71** [reply]
Hello, maybe this can help:

```
---
version: "3.3"

volumes:
  config:
  cache:
  data:

services:
  mopidy:
    image: giof71/mopidy
    container_name: mopidy
    devices:
      - /dev/snd:/dev/snd
    environment:
      - AUDIO_OUTPUT=alsasink device=hw:D10
      - RESTORE_STATE=yes
      - SCROBBLER_ENABLED=${SCROBBLER_ENABLED:-}
      - SCROBBLER_USERNAME=${SCROBBLER_USERNAME:-}
      - SCROBBLER_PASSWORD=${SCROBBLER_PASSWORD:-}
      - TIDAL_ENABLED=yes
      - TIDAL_QUALITY=${TIDAL_QUALITY:-LOSSLESS}
    ports:
      - 6680:6680
    volumes:
      - config:/config
      - cache:/cache
      - data:/data
    restart: always
```

The application would start as root so there is nothing to be done for the volumes, no need to create folders and to set permissions.
The port 8989 is not needed with the latest version. When you browse Tidal, a QR code and and address appears. I tried the address and opened it in the browser, authenticated with Tidal and authorized the device.

After that I just refrereshed the mopidy page and the plugin started working as expected.
You probably just need to correct AUDIO_OUTPUT with the name of your dac, or the index.
Let me know if you want to try it and need any help.

**Reply by Spoker1973** [reply]
Should i first to install mopidy docker and then mopidy tidal plugin or only mopidy tidal plugin?

**Reply by GioF71** [reply]
> Should i first to install mopidy docker and then mopidy tidal plugin or only mopidy tidal plugin?

Hello, you just need to use that compose file. The plugin is built in, you do not need to install it.
Let me know if this helps

**Comment by Spoker1973** [comment]
Hi GioF71, first of all Happy New Year.  I`m just thinking myself about hi res in Tidal Connect as it is in new Volumio. Maybe it is stupid but i asked chat gpt about newest version of tidal connect docker and i get some code.
There is a code with the newest binaries. Maybe you could to download those files and try to implement in your docker?
Here it is:
# Base image with minimal footprint
FROM ubuntu:latest

# Set environment variables
ENV DEBIAN_FRONTEND=noninteractive

# Update and install necessary packages
RUN apt-get update && apt-get install -y \
    curl \
    wget \
    libasound2 \
    libssl-dev \
    alsa-utils \
    python3 \
    python3-pip \
    libavahi-client3 && \
    apt-get clean && rm -rf /var/lib/apt/lists/*

# Create a directory for Tidal Connect binaries
WORKDIR /tidal-connect

# Download the latest Tidal Connect binaries
RUN wget https://repository.tidal.com/path/to/latest/tidal-connect-binary.tar.gz -O tidal-connect.tar.gz && \
    tar -xzf tidal-connect.tar.gz && rm tidal-connect.tar.gz

# Expose necessary ports
EXPOSE 80 443

# Add a script to launch Tidal Connect
COPY launch.sh /usr/local/bin/launch.sh
RUN chmod +x /usr/local/bin/launch.sh

# Set entrypoint to the launch script
ENTRYPOINT ["/usr/local/bin/launch.sh"]

**Comment by GioF71** [comment]
Hello @Spoker1973 and thank you for your effort. This could very well be done, but the problem is how to source the binary. You might already have seen that the url provided by chatgpt is not valid. AFAIK Tidal don't release the Tidal Connect app, they only provide it to select OEM.

**Reply by Spoker1973** [reply]
Indeed link provided by gptchat is not valid, my bad :( . I am testing now volumio, they let you test for free for one month. Unfortunately native tidal is only 16 bit /44.1MHz however tidal connect is provided full, you can stream even 24 / 192, i have checked it. Maybe it would be possible to extract tidal connect binaries from volumio? I could give you my credentials to connect to volumio.

**Reply by Spoker1973** [reply]
There is also one more thing. IFI Zen Stream uses volumio source code. You can find and download latest software and then try to extract hi res tidal connect? Am i wrong?

**Reply by GioF71** [reply]
> Indeed link provided by gptchat is not valid, my bad :( . I am testing now volumio, they let you test for free for one month. Unfortunately native tidal is only 16 bit /44.1MHz however tidal connect is provided full, you can stream even 24 / 192, i have checked it. Maybe it would be possible to extract tidal connect binaries from volumio? I could give you my credentials to connect to volumio.

hello, if I set the quality to HiRes instead of Highest Possible, I get hires also from native tidal with Volumio! Can you try this?

**Reply by Spoker1973** [reply]
> hello, if I set the quality to HiRes instead of Highest Possible, I get hires also from native tidal with Volumio! Can you try this?

Hello, i`ve tried and i can confirm it works in Hi Res. Nice to know it :)

**Comment by Spoker1973** [comment]
Hi GioF71 
I would like to go back to our conversation about hi res in tidal connect.
Recently I talked to a friend who has a Gustart R26 DAC. He told me that there is an instruction on github that you can use to activate tidal connect on this device. It turned out that it actually works and is based on the IFI streamer repositories. As we know, IFI and gustard are based on Debian. Here is a link to the github repository: https://gist.github.com/NoaHimesaka1873/8ef2396d5296765c37170e54c4ab1745
Maybe you can use this in your project to get hi res tidal connect. Can you take a look at it?

**Reply by GioF71** [reply]
Hello, thank you for linking me this. From the istructions it looks like we are talking about the same ~~libraries~~ binaries. That github [repo](https://github.com/shawaj/ifi-tidal-release) is from five years ago.
Is there anyone claiming that it works in hires mode?

**Reply by Spoker1973** [reply]
Personally didn`t see it works but my friend told me it works with hi res

**Reply by GioF71** [reply]
In the discussion, they talk about mConnect. That's another method and does not use the Tidal app. Is your friend positive about this?

**Reply by GioF71** [reply]
I need to say that I appreciate your effort anyway @Spoker1973!

**Reply by Spoker1973** [reply]
thank you, i do my best :) and would be glad if some day could hear hi res tidal connect :)
